openfl/federated/plan/plan.py

```diff
@@ -25,14 +40,22 @@
 DEFAULTS = 'defaults'
 AUTO = 'auto'
 
+# Type definition for an instance of a openfl Component or Federated DataLoader/TaskRunner
+Component = Any
+
+PartYAMLValueType = Union[int, float, str, bool]
+YAMLValueType = Union[PartYAMLValueType,
+                      Dict[str, Union[PartYAMLValueType, Dict[str, PartYAMLValueType]]]]
```

Comment:
```suggestion
                      Dict[str, 'YAMLValueType']]
```
The level of nesting of dictionaries here might be infinite. We can use forward referencing to specify the recursion. And everywhere below we can just use this type instead of `Optional[Dict[str, YAMLValueType]]`.